python/ray/train/_checkpoint.py

```diff
@@ -301,3 +318,24 @@ def _list_existing_del_locks(path: str) -> List[str]:
     then this should return a list of 2 deletion lock files.
     """
     return list(glob.glob(f"{_get_del_lock_path(path, suffix='*')}"))
+
+
+def _raise_migration_error(name: str):
+    raise AttributeError(
+        f"The new `ray.train.Checkpoint` class does not support `{name}()`. "
+        f"Instead, only directories are supported.\n\n"
+        f"Example to store a dictionary in a checkpoint:\n\n"
+        f"import os, pickle, tempfile\n"
+        f"from ray import train\n"
+        f"from ray.train import Checkpoint\n\n"
+        f"with tempfile.TemporaryDirectory() as checkpoint_dir:\n"
+        f"  with open(os.path.join(checkpoint_dir, 'data.json'), 'wb') as fp:\n"
+        f"    pickle.dump({{'data': 'value'}}, fp)\n\n"
+        f"  checkpoint = Checkpoint.from_directory(checkpoint_dir)\n"
+        f"  train.report(..., checkpoint=checkpoint)\n\n"
+        f"Example to load a dictionary from a checkpoint:\n\n"
+        f"if train.get_checkpoint():\n"
+        f"  with train.get_checkpoint().as_directory() as checkpoint_dir:\n"
+        f"    with open(os.path.join(checkpoint_dir, 'data.json'), 'rb') as fp:\n"
+        f"      data = pickle.load(fp)"
```

Comment:
json.load/dump

Reply:
ah I think I'll just use `data.pkl` instead. I think json may be too restrictive